org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/markers/SonarLintMarkerResolutionGenerator.java

```diff
@@ -46,4 +46,38 @@ public boolean hasResolutions(final IMarker marker) {
     }
   }
 
+  public int cognitivelyComplex(int a, int b, int c, int d) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 24 to the 15 allowed. ([squid:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.eclipse%3Asonarlint-eclipse-parent&pullRequest=147&issues=AWKzcUUtool-HGyLJXhD&open=AWKzcUUtool-HGyLJXhD)

[ISSUE_KEY]: AWKzcUUtool-HGyLJXhD